templates/pages/reset-password.jsx

```diff
@@ -121,10 +133,20 @@ var ResetPassword = React.createClass({
         window.location = Url.format(redirectObj);
         return;
       }
-      // handle errors!
+
+      this.setState({
+        expired: true,
+        submitForm: false,
+        email: false
+      });
     }.bind(this)).catch(function(ex) {
```

Comment:
is there support for `.finally(function(...) { ... })`, here, so we can house that duplicated `this.setState` in a single block?